activerecord/lib/active_record/tasks/mysql_database_tasks.rb

```diff
@@ -71,7 +74,13 @@ def structure_load(filename)
         args = prepare_command_options('mysql')
         args.concat(['--execute', %{SET FOREIGN_KEY_CHECKS = 0; SOURCE #{filename}; SET FOREIGN_KEY_CHECKS = 1}])
         args.concat(["--database", "#{configuration['database']}"])
-        Kernel.system(*args)
+        
+        case Kernel.system(*args)
+        when nil
+          raise "Unable to load structure, is mysql installed?"
+        when false
+          raise "Error loading structure, mysql exited with status #{$?.exitstatus}"
+        end
```

Comment:
Ca we extract those into a general helper as we did in `PostgreSQLDatabaseTasks`?

```ruby
      def run_cmd(cmd, args, action)
        fail run_cmd_error(cmd, args, action) unless Kernel.system(cmd, *args)
      end

      def run_cmd_error(cmd, args, action)
        msg = "failed to execute:\n"
        msg << "#{cmd} #{args.join(' ')}\n\n"
        msg << "Please check the output above for any errors and make sure that `#{cmd}` is installed in your PATH and has proper permissions.\n\n"
        msg
      end
```